pkg/env/costmodelenv.go

```diff
@@ -429,14 +429,10 @@ func IsETLEnabled() bool {
 	return GetBool(ETLEnabledEnvVar, true)
 }
 
-// GetETLMaxBatchDuration limits the window duration of the most expensive ETL
-// queries to a maximum batch size, such that queries can be tuned to avoid
-// timeout for large windows; e.g. if a 24h query is expected to timeout, but
-// a 6h query is expected to complete in 1m, then 6h could be a good value.
-func GetETLMaxBatchDuration() time.Duration {
-	// Default to 6h
-	hrs := time.Duration(GetInt64(ETLMaxBatchHours, 6))
-	return hrs * time.Hour
+func GetETLMaxPrometheusQueryDuration() time.Duration {
+	dayMins := 60 * 24
+	mins := time.Duration(GetInt64(ETLMaxPrometheusQueryDurationMinutes, int64(dayMins)))
+	return mins * time.Minute
```

Comment:
Do you think we should force this number of minutes to be a divisor of `60 * 24` or `60`? I'm wondering what will happen if we try to build in prime number windows, for example.

Reply:
It should be fine. The code just advances in steps of `max(this value, remaining interval)`, so providing a number that is relatively prime with the duration querying will just result in a final segment of a slightly smaller window. But good question!